include/mitsuba/core/distr_1d.h

```diff
@@ -233,6 +233,13 @@ template <typename Float> struct DiscreteDistribution {
         };
     }
 
+    void set_grad_suspended(bool state) {
+        if constexpr (ek::is_diff_array_v<Float>) {
```

Comment:
Here, and everywhere else in the patch, remove the `ek::is_diff_array<Float>` guard. ``ek::set_grad_suspended` expands to ..nothing.. (i.e. optimized away) when the input array is not differentiable. If you encounter any issues with that, it would be considered an Enoki bug.